### PR TITLE
SBAR-47 phase3: Add in aggregate results

### PR DESCRIPTION
**Pull request by thisisstephenbetts**
As with the previous iteration we need to provide aggregation over the specified dimensions. For expediency, I have not done all possible permutations of aggregation, but only aggregating from right to left.

For example, if we have result rows
```
[
  ['dim1_a', 'dim2_a', 'dim3_a'],
  ['dim1_a', 'dim2_a', 'dim3_b'],
  ['dim1_a', 'dim2_b', 'dim3_a'],
  ['dim1_a', 'dim2_b', 'dim3_b'],
  ['dim1_b', 'dim2_a', 'dim3_a'],
  ['dim1_b', 'dim2_a', 'dim3_b'],
  ['dim1_b', 'dim2_b', 'dim3_a'],
  ['dim1_b', 'dim2_b', 'dim3_b'],
]
```

We will add in aggregate rows
```
  ['dim1_a', 'dim2_a', 'ALL'],
  ['dim1_a', 'dim2_b', 'ALL'],
  ['dim1_b', 'dim2_a', 'ALL'],
  ['dim1_b', 'dim2_b', 'ALL'],
  ['dim1_a', 'ALL', 'ALL'],
  ['dim1_b', 'ALL', 'ALL'],
  ['ALL', 'ALL', 'ALL'],
```

We will not add in rows for
```
  ['ALL', 'ALL', 'dim3_a'],
  ['ALL', 'dim2_a', 'ALL'],
etc., etc.
```

For one thing you can already see the number of rows is becoming unwieldy. Secondly, these other permutations seem less useful. And lastly if we want different permutations we can re-run the report just specifying different orders of the report dimensions. 

The new methods to do this are fairly recursive — I've tried to add comments and use clear names, but it's not tricky. Any comments on where potential confusion arises would be very helpful.

### CODE REVIEW

**Comment by nodanaonlyzuul** [comment]
Howdy Padnah'

I believe we need to add the `-d` flag into `run-reports.sh`.

* My first instinct is that the value (often "SearchedRepo,SearchedFrom") should be an environment variable...so developers can experiment with passing different dimensions by editing their `.env` file that we so often pass as the `--env-file` tool to `docker run`.
 
* We should fix things (maybe at the ops parsing) - to not include the `:dimensions` 
option if `-d` is passed with empty string as its value.

* Document it in `.env.sample` too?

**Comment by thisisstephenbetts** [comment]
Yep, agree with those bits of feedback, thanks.

Will add them in.

File: lib/report_generators/search_term_by_dimensions.rb [hunk 1]
```diff
@@ -35,34 +37,28 @@ def report_output_path
...
-      query_response.rows.each do |query_row|
-        queries << QueryResponse.new({
-          search_term: query_row.shift,
-          action: query_row.shift,
-          dimensions: dimensions_for_query_sent.map {|dim| [dim[:name], query_row.shift]}.to_h,
-          total_events: query_row.shift.to_i,
-          unique_events: query_row.shift.to_i,
-        })
-      end
+    query_rows = get_query_rows
```
Comment (thisisstephenbetts) [inline lib/report_generators/search_term_by_dimensions.rb]: Hey @nodanaonlyzuul! Look what I did!

**Comment by nodanaonlyzuul** [comment]
Fixes https://github.com/NYPL/search-analytics-report-creator/pull/15

**Pull request merged**